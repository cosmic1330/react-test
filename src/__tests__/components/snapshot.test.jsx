import React from 'react'
import { create, act } from 'react-test-renderer'
import Library from '../../components/library'

describe('snapshot', () => {
    test('it shows the expected text when clicked', () => {
        let component
        act(() => {
            component = create(<Library />)
        })
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
        const instance = component.root

        const button = instance.findByType('button')
        const span = instance.findByType('span')
        act(() => button.props.onClick())
        tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
