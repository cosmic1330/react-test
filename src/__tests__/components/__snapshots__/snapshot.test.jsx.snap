// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot it shows the expected text when clicked 1`] = `
<div
  className="library"
>
  <span
    data-testid="span"
  >
    0
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      點我
    </button>
  </div>
</div>
`;

exports[`snapshot it shows the expected text when clicked 2`] = `
<div
  className="library"
>
  <span
    data-testid="span"
  >
    1
  </span>
  <div>
    <button
      onClick={[Function]}
    >
      點我
    </button>
  </div>
</div>
`;
